home/views.py

```diff
@@ -56,13 +56,16 @@ class EventListView(ListView):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
-        events = Event.objects.visible().order_by("-start_time")
+        upcoming_events = Event.objects.visible().upcoming().order_by("-start_time")
+        past_events = Event.objects.visible().past().order_by("-start_time")
 
         tag = self.request.GET.get("tag")
         if tag:
-            events = events.filter(tags__name=tag)
+            upcoming_events = upcoming_events.filter(tags__name=tag)
+            past_events = past_events.filter(tags__name=tag)
 
-        context["events"] = events
+        context["upcoming_events"] = upcoming_events
+        context["past_events"] = past_events
```

Comment:
We could use the base `events` queryset, then apply the datetime filtering just before it's needed:

```suggestion
        context["upcoming_events"] = events.upcoming()
        context["past_events"] = events.past()
```

Reply:
Thank you ⭐